compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6299,25 +6299,25 @@ private BType getCompletionType(BType collectionType, BLangQueryExpr queryExpr,
     }
 
     private BType getNonContextualQueryType(BType staticType, BType basicType) {
```

Comment:
The problem here is we are not getting the actual basic type
From the spec https://ballerina.io/spec/lang/2022R1/#query-expr:
> If there is no contextually expected type, then the basic type of the value constructed is the basic type of expression following in; it is a compile-time error if the static type of this expression is not a subtype of one of the basic types that a query expression can construct.

Also, returning array type as the result type for all other cases also incorrect which would be a breaking change. Hence, please create a separate issue for that and may be handle it in this same PR

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/37670. @SasinduDilshara can you please have a look